lib/libxdp/libxdp.c

```diff
@@ -2130,9 +2157,10 @@ static int xdp_multiprog__link_prog(struct xdp_multiprog *mp,
 static struct xdp_multiprog *xdp_multiprog__generate(struct xdp_program **new_progs,
```

Comment:
The name new_progs is really confusing, not sure what would be better though...  Maybe chg_progs? 

So thinks like ```if (!remove_new)``` look really odd. What are we doing are we not removing_new?! But you mean new_progs is not used as new but as remove...

But I'm ok with new_progs, but maybe this function needs a comment?